types/jest/index.d.ts

```diff
@@ -757,8 +758,25 @@ declare namespace jest {
         toMatch(expected: string | RegExp): R;
         /**
          * Used to check that a JavaScript object matches a subset of the properties of an object
+         *
+         * Optionally, you can provide an object to use as Generic type for the expected value.
+         * This ensures that the matching object matches the structure of the provided object-like type.
```

Comment:
I don't see any comments similar to this in here.

I believe the documentation throughout the file is intended to document Jest's API as stated in its docs, not on how to use the types. The type information, such as generic arguments, can still be inferred from the editor sometimes:

![Screen Shot 2019-04-27 at 11 28 34 AM](https://user-images.githubusercontent.com/2100222/56851834-88966a00-68e1-11e9-9981-21a7d67d47f0.png)

Reply:
I based the including of an example on `.toHaveNthReturnedWith()` typing just above `toMatchObject()`. As for inferring from IntelliSense, the examples are in fact also included in the IntelliSense and I personally thing it's nice to have some extra documentation, especially for TypeScript structures that are considered "advanced" (being that Generics are under "advanced types" on the docs) so it helps TS starters.

GIF recording of the IntelliSense I see, with the example:

![19-57-46_27-04-2019_%pn](https://user-images.githubusercontent.com/4019718/56853369-04ff6c00-6927-11e9-86a1-5861a14434ae.gif)


TL;DR: I think it doesn't give any negative effects and only (if minor), positive ones. I see no reason to remove it.

Reply:
The comment added stood out a little bit to me from the rest of the documentation comments since it isn't describing Jest's functionality, but given what you said above I don't think it hurts to keep it either. Thanks for the clarification! 😄